### PR TITLE
Style Variation: Use global-styles endpoint to apply the selected variation

### PR DESCRIPTION
**Pull request by arthur791004**
#### Proposed Changes

* This PR is focusing on switching to the `global-styles` endpoints to apply the variation.
* Note that the `global-styles` endpoints only support the active theme for now, so we have to do it after we activated the theme.

#### Testing Instructions

* Head to the logged-in Theme Showcase `/themes/${your_site}`. If using calypso.live, the flag `themes/showcase-i4/details-and-preview` is required.
* Find a theme with style variation, open the theme preview modal, and select one of the styles in the sidebar.
* Activate the theme and ensure that your site has the theme with the selected style applied.

Note that the selected style variation should be applied successfully on both simple site and atomic site 🙂

#### Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to https://github.com/Automattic/wp-calypso/pull/71754

### CODE REVIEW

**Comment by taipeicoder** [comment]
@arthur791004 I couldn't access the calypso.live build and the E2E fails when re-triggered. Could you rebase this and see if that fixes things? Thanks.

**Comment by arthur791004** [comment]
Rebased!

File: client/state/themes/actions/activate-theme.js [hunk 1]
```diff
@@ -46,7 +51,24 @@ export function activateTheme(
 				...( isEnabled( 'themes/theme-switch-persist-template' ) && {
 					persist_homepage_template: true,
 				} ),
-				...( styleVariationSlug && { style_variation_slug: styleVariationSlug } ),
+			} )
+			.then( async ( theme ) => {
+				if ( styleVariationSlug ) {
+					const themeStylesheet = theme.stylesheet || themeId;
+					const globalStylesId = await dispatch( getGlobalStylesId( siteId, themeStylesheet ) );
```
Comment (taipeicoder) [inline client/state/themes/actions/activate-theme.js]: [nit] We can move this line to:
```js
if ( currentVariation ) {
     const globalStylesId = await dispatch( getGlobalStylesId( siteId, themeStylesheet ) );
     await dispatch( updateGlobalStyles( siteId, globalStylesId, currentVariation ) );
}
```

What do you think?

File: client/state/global-styles/actions.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			apiNamespace: 'wp/v2',
+		} );
+
+		return variations;
+	};
+}
+
+export function updateGlobalStyles(
+	siteSlug: string,
+	globalStyleId: number,
```
Comment (taipeicoder) [inline client/state/global-styles/actions.ts]: [nit] Can we rename `globalStyleId` to `globalStylesId` for consistency?

File: packages/data-stores/src/site/actions.ts [hunk 3]
```diff
@@ -208,6 +209,52 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 		return globalStyles;
 	}
 
+	function* setGlobalStyles( siteSlug: string, globalStyleId: number, globalStyles: GlobalStyles ) {
```
Comment (taipeicoder) [inline packages/data-stores/src/site/actions.ts]: [nit] Same here, `globalStylesId` 🙂

File: client/state/global-styles/actions.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import wpcom from 'calypso/lib/wp';
+import type { ActiveTheme, GlobalStyles } from '@automattic/data-stores';
+
+export function getGlobalStylesId( siteSlug: string, stylesheet: string ) {
```
Comment (taipeicoder) [inline client/state/global-styles/actions.ts]: Seems that we can either pass `siteId` or `siteSlug` here. Can we support either?

**Comment by taipeicoder** [comment]
We'll need to update this test as well https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/site/test/actions.ts#L451

File: client/state/themes/actions/activate-theme.js (hunk 1 above)
Comment (arthur791004) [reply client/state/themes/actions/activate-theme.js]: Sounds good 👍

**Comment by taipeicoder** [comment]
I'm seeing the following API response for `https://public-api.wordpress.com/wp/v2/sites/${SITE_ID}/global-styles/themes/pixl/variations` 🥲: 
Screen Shot 2023-01-18 at 5 35 27 PM

EDIT: Relevant conversatino p1669895754407499/1669884996.711509-slack-C7YPUHBB2

**Comment by arthur791004** [comment]
It's expected. We have to add it to the mappers

**Comment by arthur791004** [comment]
@taipeicoder You can apply D98662-code to your sandbox and test it on the Jetpack site again!

**Review by taipeicoder — APPROVED** [review]
Works great after applying D98662-code!

**Pull request merged**